tck/src/main/java/org/eclipse/microprofile/concurrency/tck/ThreadContextTest.java

```diff
@@ -580,4 +582,161 @@ public void thirdPartyContextProvidersAreIncludedInThreadContext() {
             Thread.currentThread().setPriority(originalPriority);
         }
     }
+    
+     /**
+     * Verify the MicroProfile Concurrency implementation of propagate(), cleared(), and unchanged()
+     * for ThreadContext.Builder.
+     */
+    @Test
+    public void contextControlsForThreadContextBuilder() throws InterruptedException, ExecutionException, TimeoutException {
+    	ThreadContext allContext = ThreadContext.builder()
+    			.propagated(Buffer.CONTEXT_NAME)
+    			.cleared(Label.CONTEXT_NAME)
+    			.unchanged(THREAD_PRIORITY)
+    			.build();
+    	
+    	try {
+        	ThreadContext.builder()
+			.propagated(Buffer.CONTEXT_NAME)
+			.cleared(Label.CONTEXT_NAME, Buffer.CONTEXT_NAME)
+			.unchanged(THREAD_PRIORITY)
+			.build();
+        	Assert.fail("ThreadContext.Builder.build() should throw an IllegalStateException for set overlap between propagated and cleared");
+    	} catch (IllegalStateException ISE) {
+    		//expected.
+    	}
+
+        int originalPriority = Thread.currentThread().getPriority();
+        try {
+            // Set non-default values
+            int newPriority = originalPriority == 4 ? 3 : 4;
+            Buffer.get().append("contextControls-test-buffer-A");
+            Label.set("contextControls-test-label-A");
+
+            //Future<Integer> future1 = unmanagedThreads.submit(
+            		
+            Callable<Integer> callable = allContext.contextualCallable(() -> {
+            	Assert.assertEquals(Buffer.get().toString(), "contextControls-test-buffer-A-B",
+            			"Context type was not propagated to contextual action.");
+
+            	Buffer.get().append("-C");
+
+            	Assert.assertEquals(Label.get(), "",
+            			"Context type that is configured to be cleared was not cleared.");
+
+            	Label.set("contextControls-test-label-C");
+            	
+            	return Thread.currentThread().getPriority();
+            });
+            
+            Buffer.get().append("-B");
+            Label.set("contextControls-test-label-B");
+
+            Future<Integer> future = unmanagedThreads.submit(() -> {
+            		Buffer.get().append("unpropagated-buffer");
+            		Label.set("unpropagated-label");
+            		Thread.currentThread().setPriority(newPriority);
```

Comment:
You need a finally block to clean up each of these three contexts that you are setting on the unmanaged thread. Because it is an unmanaged thread, there is nothing to clean it up for you, so the context of this thread will remained corrupted when used by other tests at some future point.